pkg/repo/repotest/server.go

```diff
@@ -36,11 +37,13 @@ import (
 //
 // The caller is responsible for destroying the temp directory as well as stopping
 // the server.
-func NewTempServer(glob string) (*Server, error) {
+func NewTempServer(t *testing.T, glob string) (*Server, error) {
```

Comment:
As this is a public function signature, we cannot change it due to backwards compatibility concerns as per our contributing guide (see CONTRIBUTING.md).

I would suggest instead introducing a new function called `NewTempServerWithCleanup(t *testing.T, glob string) (*Server, error) {}`, and add a comment above `NewTempServer` that this function has been deprecated in favour of `NewTempServerWithCleanup`. I'd also add an additional comment to replace `NewTempServer` with `NewTempServerWithCleanup` in Helm 4.

Thanks!

Reply:
Yeap, great thanks for your suggestion.